src/Program/Model/LuciConnection.hs

```diff
@@ -104,7 +104,7 @@ luciBehavior lsettings geoJSONImportFire cityB groundUpdatedE
       -- asking luci to save a scenario on button click
       (askSaveScenarioE, onAskSaveScenarioFire) <- newEvent
       liftIO $ GUI.registerSaveScenario onAskSaveScenarioFire
-      scenarioSavedE <- runScenarioCreate luciClientB $ (\ci s -> (s, originLonLatAlt ci, srid ci, storeCityAsIs ci)) <$> cityB <@> askSaveScenarioE
+      scenarioSavedE <- runScenarioCreate luciClientB $ (\ci s -> (s, originLonLatAlt ci, srid ci, storeCityAsIs ci, ci)) <$> cityB <@> askSaveScenarioE
```

Comment:
If you pass city itself, there is no need to pass its properties anymore. originLonLatAlt ci, srid ci, storeCityAsIs ci should go away - you can get them inside `runScenarioCreate` function